### PR TITLE
Octavia

### PR DESCRIPTION
**Pull request by jtopjian**
For #852

### CODE REVIEW

**Comment by jtopjian** [comment]
AFAICT, this works. I ran the acceptance tests on vexxhost and everything worked. There's an issue with the `TestLoadbalancersCRUD` test where the call to cascading delete needs to wait until everything is deleted before proceeding to delete the network and subnet, but that's not a blocking/breaking issue.

**Comment by lingxiankong** [comment]
seems the go fmt failed

**Comment by lingxiankong** [comment]
looks good to me now, thanks!

**Comment by jtopjian** [comment]
@lingxiankong thanks!

@jrperritt when you have a moment, can you do a quick review? I think this is good to go.

File: openstack/loadbalancer/v2/loadbalancers/requests.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+func Delete(c *gophercloud.ServiceClient, id string) (r DeleteResult) {
+	_, r.Err = c.Delete(resourceURL(c, id), nil)
+	return
+}
+
+// CascadingDelete is like `Delete`, but will also delete any of the load balancer's
+// children (listener, monitor, etc).
+// NOTE: This function will only work with Octavia load balancers; Neutron does not
+// support this.
+func CascadingDelete(c *gophercloud.ServiceClient, id string) (r DeleteResult) {
```
Comment (jrperritt) [inline openstack/loadbalancer/v2/loadbalancers/requests.go]: Did we want to instead add a `DeleteOptsBuilder` to `Delete`?

Comment (jtopjian) [reply openstack/loadbalancer/v2/loadbalancers/requests.go]: Good point. Yes. Better to do it now.

The recent set of commits have this added, though I had to do a force-push due to the merge conflict.

**Comment by jtopjian** [comment]
@jrperritt ping for review

**Pull request merged**